docs/notebooks/.ipynb_checkpoints/WMD_tutorial-checkpoint.ipynb

```diff
@@ -0,0 +1,601 @@
+    "        \n",
+    "        if json_line['business_id'] == ids[0]:\n",
+    "            # Only use this restaurant in corpus for WmdSimilarity.\n",
+    "            wmd_corpus.append(text)\n",
+    "            documents.append(json_line['text'])\n",
+    "\n",
+    "print '\\nCell took %f seconds to run.' %(time() - start)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Train Word2Vec on all the restaurants.\n",
+    "model = Word2Vec(w2v_corpus, workers=7, size=100)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we want to initialize the similarity class with a corpus and a word2vec model (which provides the embeddings and the `wmdistance` method itself)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Cell took 158.570173 seconds to run.\n"
+     ]
+    }
+   ],
+   "source": [
+    "start = time()\n",
+    "\n",
+    "from gensim.similarities import WmdSimilarity\n",
+    "instance = WmdSimilarity(wmd_corpus, model, num_best=10)\n",
+    "\n",
+    "print 'Cell took %f seconds to run.' %(time() - start)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `num_best` parameter decides how many results the queries return. Now let's try making a query. The output is a list of indeces and similarities of documents in the corpus, sorted by similarity. Note that the output format is slightly different when `num_best` is `None` (i.e. not assigned).\n",
+    "\n",
+    "The query below is taken directly from one of the reviews in the corpus. Let's see if there are other reviews that are similar to this one."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "sent = 'Very good, you should seat outdoor.'\n",
+    "\n",
+    "# Pre-process query.\n",
+    "query = sent.lower()  # Lower the text.\n",
+    "query = word_tokenize(query)  # Split into words.\n",
+    "query = [w for w in query if not w in stop_words]  # Remove stopwords.\n",
+    "query = [w for w in query if w.isalpha()]  # Remove numbers and punctuation.\n",
+    "\n",
+    "sims = instance[query]  # A query is simply a \"look-up\" in the similarity class."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The query and the most similar documents, together with the similarities, are printed below. We see that the retrieved documents are discussing the same thing as the query, although using different words. The query talks about getting a seat \"outdoor\", while the results talk about sitting \"outside\", and one of them says the restaurant has a \"nice view\"."
```

Comment:
Please print all 10 similar documents as we asked for num_best=10 and it is confusing we only get 3